### PR TITLE
Update docs after contract simplifications

### PR DESCRIPTION
**Pull request by fredo**
Main changes:

- Request manager / Resolution registry merge
- removal of fill hash
- new RestrictedCalls interface

resolves: #1033

As a reviewer, please download the branch, read the full docs, and comment on anything which is not understandable.

### CODE REVIEW

File: docs/source/l1_resolution.rst [hunk 1]
```diff
@@ -25,11 +25,11 @@ on L1.
 
 The central role for routing this information correctly has the ``Resolver``
 contract, which is deployed on the shared L1 chain of both rollups. The resolver
-holds a mapping of chain ids to the contract address of the ``ResolutionRegistry``
+holds a mapping of chain ids to the contract address of the ``RequestManager``
 on that chain.
 
 When a proof transaction sent by the target rollup is executed, the resolver can
-find the correct registry in the mapping and forward the information about the
+find the correct manager in the mapping and forward the information about the
```
Comment (istankovic) [inline docs/source/l1_resolution.rst:32]:
```suggestion
find the correct request manager in the mapping and forward the information about the
```

File: docs/source/whitepaper.rst [hunk 2]
```diff
@@ -232,22 +232,23 @@ L1 resolutions
...
 
 To trigger L1 resolution is to apply this call on L1 using the data from the rollup B's outbox. This will forward the
-information from the resolver to the inbox of rollup A in the form of a call to the ``resolution registry``.
-This registry will store in its state a mapping from ``fill hash`` to ``Bob``, allowing the ``request manager``
-to verify that a claim to fill a certain request with a certain fill ID is honest. Rollup A's chain ID is necessary for
-the ``resolver`` contract to know to which ``resolution registry`` to forward the proof to. Rollup B's chain ID is used to
+information from the resolver to the inbox of rollup A in the form of a call to the ``request manager``.
+This request manager will store ``Bob`` as the rightful filler and the ``fill ID`` to the request object. This marks any
+claim created by Bob with the corresponding ``fill ID`` to be valid. The  Rollup A's chain ID is necessary for
+the ``resolver`` contract to know to which ``request manager`` to forward the proof to. Rollup B's chain ID is used to
```
Comment (istankovic) [inline docs/source/whitepaper.rst:245]:
```suggestion
the ``Resolver`` contract to know to which ``request manager`` to forward the proof to. Rollup B's chain ID is used to
```

File: docs/source/whitepaper.rst [hunk 3]
```diff
@@ -232,22 +232,23 @@ L1 resolutions
...
 restrict the call to authenticated ``fill manager`` and ``cross domain messenger`` contracts.
```
Comment (istankovic) [inline docs/source/whitepaper.rst:246]:
```suggestion
restrict the call to authenticated messenger contract on rollup B.
```

File: docs/source/whitepaper.rst [hunk 4]
```diff
@@ -362,18 +363,17 @@ was not the correct filler. However, Alice's request might not be able to be fil
...
-It then submits a proof to the outbox of rollup B indicating that the fill hash is invalid, i.e. that the request ID
-cannot be mapped to the fill hash.
+When called, the fill manager contract on rollup B checks that no fills exists for the corresponding request ID and fill ID.
+It then submits a proof to the outbox of rollup B indicating that the fill ID is invalid for the given request ID, i.e.
+that the request ID cannot be mapped to the fill ID.
 
 Similarly to the filled L1 resolution case, Bob can then trigger a call on L1 to forward this message to rollup A. This
-message will store a flag in the ``resolution registry`` stating that the ``fill hash`` is invalid. This invalidates any
-claim with the corresponding ``fill hash``.
+message will store a flag in the ``request manager`` stating that the fill ID is invalid for the given request. This
```
Comment (istankovic) [inline docs/source/whitepaper.rst:371]:
```suggestion
message will store a flag in the request manager stating that the fill ID is invalid for the given request. This
```

File: docs/source/whitepaper.rst [hunk 5]
```diff
@@ -362,18 +363,17 @@ was not the correct filler. However, Alice's request might not be able to be fil
...
+invalidates any claim with the corresponding fill ID.
 
 To make sure the proof arrives in time on rollup A, Bob will need to call the ``fill manager`` as soon as he notices a
```
Comment (istankovic) [inline docs/source/whitepaper.rst:374]:
```suggestion
To make sure the proof arrives in time on rollup A, Bob will need to call the fill manager as soon as he notices a
```

File: docs/source/whitepaper.rst (hunk 4 above)
Comment (fredo) [reply docs/source/whitepaper.rst:371]: Can you read the whole section, please? All other contracts are highlighted as well. Should I remove this from all of them?

**Review by istankovic — APPROVED** [review]
Nice!

**Pull request merged**